### DISCUSSION TITLE
-level black_point{,white_point}{%}{,gamma}

### DISCUSSION THREAD

**Discussion by ssbroad** [ImageMagick/ImageMagick#7265] [Help] [answered]
### ImageMagick version

7.1.1

### Operating system, version and so on

windows10

### Description

I want to use ImageMagick to adjust the contrast of a manga image. 

However, I found that using the parameter "-level 63,248,1.0", did not handle the image correctly. 
I got a pure white picture. 

When it was changed to "-level 24%,97%,1.0", it was handled correctly. 

_"Given one, two or three values delimited with commas: black-point, white-point, gamma (for example: 10,250,1.0 or 2%,98%,0.5)."_

The document is allowed without the ‘%’, but actually gets the wrong picture.

Why?

**Comment by snibgo** [comment, marked as answer]
The documentation at https://imagemagick.org/script/command-line-options.php#level says:

> Given one, two or three values delimited with commas: black-point, white-point, gamma (for example: 10,250,1.0 or 2%,98%,0.5). The black and white points range from 0 to QuantumRange, or from 0 to 100%;  ...

So if you don't use percent %, the numbers are in the range from 0 to QuantumRange. If your IM is Q8, QuantumRange is 255. For Q16, it is 65535. For Q32, it is 4294967295.

`magick -version` will show your Q-number. To see you QuantumRange, you can:
```
magick xc: -precision 15 -format %[fx:QuantumRange] info:
```
